### PR TITLE
Vault integration: add dynamic credential type to sessions

### PR DESCRIPTION
**Pull request by mgaffney**

### CODE REVIEW

File: internal/db/schema/migrations/postgres/10/04_session.up.sql [hunk 1]
```diff
@@ -1,37 +1,38 @@
...
       constraint target_credential_purpose_fkey
         references target_credential_purpose_enm (name)
         on delete restrict
         on update cascade,
     create_time wt_timestamp,
-    primary key(session_id, credential_id, credential_library_id, target_credential_purpose)
+    primary key(session_id, credential_id, library_id),
+    constraint session_credential_dynamic_library_id_credential_id_uq
+      unique(library_id, credential_id),
+    constraint session_credential_dynamic_credential_id_uq
```
Comment (louisruch) [inline internal/db/schema/migrations/postgres/10/04_session.up.sql:25]: why not define this constraint below line 4 `credential_id wt_public_id not null,`

File: internal/session/dynamic_credential_test.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			wantErr: errors.InvalidParameter,
+		},
+		{
+			name: "valid",
+			args: args{
+				sessionId:    "abcd_OOOOOOOOOO",
+				credentialId: "cred_OOOOOOOOOO",
+				library: &target.CredentialLibrary{
+					CredentialLibrary: &store.CredentialLibrary{
+						TargetId:            "target_1",
```
Comment (louisruch) [inline internal/session/dynamic_credential_test.go:98]: Doesn't seem as if `targetId` is ever used here, no harm keeping it I guess...

File: internal/target/repository.go [hunk 3]
```diff
@@ -649,3 +649,15 @@ func (r *Repository) SetTargetHostSets(ctx context.Context, targetId string, tar
 	}
 	return currentHostSets, totalRowsAffected, nil
 }
+
+func fetchLibraries(ctx context.Context, r db.Reader, targetId string) ([]*CredentialLibrary, error) {
```
Comment (louisruch) [inline internal/target/repository.go:653]: So close to us being perfectly in-sync: 
Screen Shot 2021-05-06 at 2 19 52 PM

**Review by louisruch — APPROVED** [review]
Couple comments but overall LGTM

File: internal/db/schema/migrations/postgres/10/04_session.up.sql (hunk 1 above)
Comment (mgaffney) [reply internal/db/schema/migrations/postgres/10/04_session.up.sql:25]: I was grouping the unique constraints together.

**Pull request merged**